src/core/scene/a-scene.js

```diff
@@ -139,6 +142,12 @@ class AScene extends AEntity {
     });
 
     this.initSystems();
+    // Compute component order
+    this.componentOrder = solveOrder(components, this.componentOrder);
+    this.addEventListener('componentregistered', function () {
+      // Recompute order
+      self.componentOrder = solveOrder(components, self.componentOrder);
```

Comment:
`solveOrder` not super descriptive. `sortComponents`?

Reply:
It's not sorting, the `components` object isn't mutated in the process. It just returns an order which it arrives at by solving the set of `before/after` constraints, hence `solveOrder`.

Perhaps something like `determineOrder` (or `determineComponentOrder`) could work?

Reply:
`sortComponentsByTikTokOrder` ? It's not a sort in place but still a sort

Reply:
Not trying to be pedantic, but it really isn't sorting, neither in-place nor as its result. The resulting order could be used for sorting, but isn't. Naming it as such feels out of place to me.